src/plugins/intel_npu/src/backend/src/zero_remote_tensor.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "zero_remote_tensor.hpp"
+
+#include <ze_api.h>
+
+#include "intel_npu/al/config/common.hpp"
+#include "openvino/core/type/element_iterator.hpp"
+#include "openvino/runtime/intel_npu/remote_properties.hpp"
+#include "zero_utils.hpp"
+
+namespace {
+
+constexpr std::size_t STANDARD_PAGE_SIZE = 4096;
+
+}  // namespace
+
+namespace intel_npu {
+
+ZeroRemoteTensor::ZeroRemoteTensor(std::shared_ptr<ov::IRemoteContext> context,
+                                   std::shared_ptr<ZeroInitStructsHolder> init_structs,
+                                   const ov::element::Type& element_type,
+                                   const ov::Shape& shape,
+                                   const Config& config,
+                                   bool is_input,
+                                   RemoteTensorType mem_type,
+                                   void* mem)
+    : RemoteTensor(context, element_type, shape),
+      _config(config),
+      _logger("ZeroRemoteContext", _config.get<LOG_LEVEL>()),
+      _init_structs(init_structs),
+      _is_input(is_input),
+      _mem_type(mem_type),
+      _mem(mem) {
+    _ze_properties.stype = ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES;
+    zeroUtils::throwOnFail("zeDeviceGetProperties", zeDeviceGetProperties(_init_structs->getDevice(), &_ze_properties));
+
+    const auto byte_size = ov::element::get_memory_size(_element_type, shape_size(_shape));
+
+    allocate(byte_size);
+}
+
+ZeroRemoteTensor::~ZeroRemoteTensor() {
+    auto res = deallocate();
+    if (!res) {
+        _logger.error("ZeroRemoteTensor failed to free the memory");
+    }
+}
+
+bool ZeroRemoteTensor::deallocate() noexcept {
+    switch (_mem_type) {
+    case RemoteTensorType::L0_INTERNAL_BUF:
+    case RemoteTensorType::SHARED_BUF: {
+        auto result = zeMemFree(_init_structs->getContext(), _data);
+        if (ZE_RESULT_SUCCESS != result) {
+            return false;
+        }
+        _data = nullptr;
+        return true;
+        break;
```

Comment:
Not sure if this `break` is needed as there is a `return` one line above.

Reply:
done, thanks!